packages/forms-web-app/src/dynamic-forms/validator/conditional-required-validator.js

```diff
@@ -0,0 +1,53 @@
+const { body } = require('express-validator');
```

Comment:
Another option for doing this might be to extend the existing stringValidator constructorsso we can override the field name to be validated to point at the conditional field instead, so instead of defaulting to use the questionObj.fieldName we instead use the passed in value. I think the below should work.

`const { body } = require('express-validator');
const BaseValidator = require('./base-validator.js');

class StringValidator extends BaseValidator {
	minLength = {
		minLength: 0,
		minLengthMessage: ''
	};
	maxLength = {
		maxLength: 0,
		maxLengthMessage: ''
	};
	regex = {
		regex: '',
		regexMessage: ''
	};
	constructor({ minLength, maxLength, regex, fieldName } = {}) {
		super();

		if (!minLength && !maxLength && !regex)
			throw new Error('String validator is invoked without any validations set!');
		if (minLength && minLength.minLength) this.minLength = minLength;
		if (maxLength && maxLength.maxLength) this.maxLength = maxLength;
		if (regex && regex.regex) this.regex = regex;
		this.minLength.minLengthMessage = this.minLength.minLengthMessage
			? this.minLength.minLengthMessage
			: `Input too short - Please enter at least ${this.minLength.minLength} characters`;
		this.maxLength.maxLengthMessage = this.maxLength.maxLengthMessage
			? this.maxLength.maxLengthMessage
			: `Input too long - Please enter no more than ${this.maxLength.maxLength} characters`;
		this.regex.regexMessage = this.regex.regexMessage
			? this.regex.regexMessage
			: 'Please enter only the allowed characters';
			this.fieldName = fieldName;
	}
	validate(questionObj) {
		const minLengthValidator = (chain) =>
			chain
				.isLength({ min: this.minLength.minLength })
				.withMessage(this.minLength.minLengthMessage);
		const maxLengthValidator = (chain) =>
			chain
				.isLength({ max: this.maxLength.maxLength })
				.withMessage(this.maxLength.maxLengthMessage);
		const regexValidator = (chain) =>
			chain.matches(new RegExp(this.regex.regex)).withMessage(this.regex.regexMessage);

		let chain = body(this.fieldName ? this.fieldName : questionObj.fieldName);
		if (this.minLength.minLength) {
			chain = minLengthValidator(chain);
		}
		if (this.maxLength.maxLength) {
			chain = maxLengthValidator(chain);
		}
		if (this.regex.regex) {
			chain = regexValidator(chain);
		}
		return chain;
	}
}

module.exports = StringValidator;`

This is a modified version of https://github.com/Planning-Inspectorate/appeal-planning-decision/blob/main/packages/forms-web-app/src/dynamic-forms/validator/string-validator.js which already does max length validation based on configuration.